### PR TITLE
Update callbacks when component receives new props

### PR DESCRIPTION
**Pull request by jamesdphillips**
For my use case I need to be able to pass up a new `onfinished` callback on an interval.

### CODE REVIEW

**Comment by leoasis** [comment]
Hey! First of all thanks for the PR and for finding this useful!

Regarding the change, I think a better solution for this would be to hold on the instance reference and whenever the sound calls an event, use that instance to get the latest props, in here https://github.com/leoasis/react-sound/blob/master/src/index.js#L124.

Basically the idea would be to do this:

``` js
    const instance = this;

    this.stopCreatingSound = createSound({
      url: this.props.url,
      volume: props.volume,
      whileloading() {
        instance.props.onLoading(this);
      },
      whileplaying() {
        instance.props.onPlaying(this);
      },
      onfinish() {
        instance.props.onFinishedPlaying();
      }
    }, sound => {
      this.sound = sound;
      callback(sound);
    });
```

Would you like to change the PR to do something like this?

**Comment by jamesdphillips** [comment]
That seems like a great idea. I'll take a crack at it next week 👍

**Comment by leoasis** [comment]
Awesome! Thanks for contributing! 🎉

**Comment by leoasis** [comment]
Hey @jamesdphillips, were you able to make this happen? Willing to send a PR? 😄

File: src/index.js [hunk 1]
```diff
@@ -109,6 +109,10 @@ export default class Sound extends React.Component {
       if (this.props.volume !== prevProps.volume) {
         sound.setVolume(this.props.volume);
       }
+
+      sound.options.onfinish = () => this.props.onFinishedPlaying() || null;
+      sound.options.whileplaying = () => this.props.onPlaying(this) || null;
+      sound.options.whileloading = () => this.props.onLoading(this) || null;
```
Comment (peinguin) [inline src/index.js:115]: why not just
sound.options.onfinish = this.props.onFinishedPlaying
is "this" really needed in arguments?

Comment (peinguin) [reply src/index.js:115]: I saw the code.
It will be wrong this.
Use sound instead of this.

**Review by peinguin — CHANGES_REQUESTED** [review]
It will be wrong this.
Use sound instead of this.

**Pull request closed without merging**